### DISCUSSION TITLE
Group relationships by tables

### DISCUSSION THREAD

**Discussion by MartinNordestgaard** [TabularEditor/TabularEditor3#1280] [Ideas]
### Is your feature request related to a problem? Please describe

Hi

I'm working on a pretty big data model, where I have often thought that it would be awsome to be able to obtain a greater overview of the relationships in TE2/TE3 if my relationships could be grouped - e.g. by table (either viewed from the right side or the left side of the relation).

### Describe the solution you'd like

When browsing (folding out) the relationsships in the TOM Explorer to hieracally see the relationships grouped by tables. When you fold out "a table" the individually relationsships can be seen.

### Goals

1. Better overview of relationsships
2. Easier navigation both in terms of relationsships, but also when navigating from the model object to eg. tables (my scroll wheel is hurt some times ;-) )
3. More reel estate / space in the TOM Exporer, when navigating in (big) data models

**Comment by mlonsk** [comment]
Hi @MartinNordestgaard 

I can see the utility in a better grouping of relationships. Relationships are currently ordered according to the 'From Table'. Is that what you envision for the grouping or what would make sense for you?
Additionally, I wonder if this should be feature that is not default but something to turn on. If you have a rather small model it might be more annoying having to fold out relationships.

**Reply by MartinNordestgaard** [reply]
Hi @mlonsk 

Yes, that is exactly, what I was looking for.

The grouping on the FromTable would solve my request, but I could also see the usecase for grouping on the ToTable. Maybe somekind of right click view-option, where it should default be set to "off/no grouping", as you suggest.

**Reply by Remko1981** [reply]
At the moment i never open the relationships folder and instead start a diagram and then add all tables with a relation. 

Would it not be better to show the relationships also on the table as you can see partitions? Or make it an option to move them there. 
So there would be a expandable item on the table and you can disable it from the object explorer as we can with partitions. Most of the time when i would want to view the relations i am at the object in the object explorer.

**Reply by mlonsk** [reply]
Hi @Remko1981 
The Dependencies window allows for seeing the relationship that starts from a chosen table. Do you think that helps solve your use case? 
You can dock it next to or below the TOM Explorer and enable Track TOM Explorer to always see the relationships from a chosen table or column.

![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/6800c082-29af-4187-80e9-172eb8bb2200)
